plugins/inputs/execd/execd_test.go

```diff
@@ -0,0 +1,90 @@
+// +build !windows
+
+package execd
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf/agent"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/models"
+
+	"github.com/influxdata/telegraf/plugins/parsers"
+
+	"github.com/influxdata/telegraf"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExternalInputWorks(t *testing.T) {
+	jsonParser, err := parsers.NewInfluxParser()
+	assert.NoError(t, err)
+
+	e := &Execd{
+		Command:      []string{shell(), fileShellScriptPath()},
+		RestartDelay: config.Duration(5 * time.Second),
+		parser:       jsonParser,
+		Signal:       "STDIN",
+	}
+
+	metrics := make(chan telegraf.Metric, 10)
+	defer close(metrics)
+	acc := agent.NewAccumulator(&TestMetricMaker{}, metrics)
+
+	assert.NoError(t, e.Start(acc))
+	assert.NoError(t, e.Gather(acc))
+	e.Stop()
+
+	// grab a metric and make sure it's a thing
+	m := <-metrics
+
+	assert.Equal(t, "counter_bash", m.Name())
+	val, ok := m.GetField("count")
+	assert.True(t, ok)
+	assert.Equal(t, float64(0), val)
+	// test that a later gather will not panic
+	e.Gather(acc)
+}
+
+func fileShellScriptPath() string {
+	return filepath.Join(telegrafPath(), "plugins/inputs/execd/examples/count.sh")
+}
+
+func telegrafPath() string {
+	wd, err := os.Getwd()
```

Comment:
`go test` always run from the test directory (plugins/inputs/execd), you can take advantage of this as there are many tests using this property now.  This was recommended by Dave Cheney: https://dave.cheney.net/2016/05/10/test-fixtures-in-go